internal/provider/resource_vpc_router.go

```diff
@@ -0,0 +1,292 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework-timeouts/resource/timeouts"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/oxidecomputer/oxide.go/oxide"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource              = (*vpcRouterResource)(nil)
+	_ resource.ResourceWithConfigure = (*vpcRouterResource)(nil)
+)
+
+// NewVpcRouterResource is a helper function to simplify the provider implementation.
+func NewVpcRouterResource() resource.Resource {
+	return &vpcRouterResource{}
+}
+
+// vpcRouterResource is the resource implementation.
+type vpcRouterResource struct {
+	client *oxide.Client
+}
+
+type vpcRouterResourceModel struct {
```

Comment:
Same question about adding `kind` to this from earlier although `kind` for created VPC routers will always be `custom` since it's not possible to create a `system` VPC router to my knowledge.

Reply:
yeah, we should add it definitely thanks!